plugins/modules/omevv_firmware.py

```diff
@@ -0,0 +1,1224 @@
+        checks if the update job is allowed, and handles the firmware update process.
+
+        Returns:
+            None
+        """
+        cluster_name = None
+        cluster_group_id = None
+        host_service_tags = None
+        new_host_id = None
+        payload = None
+        self.validate_params()
+        vcenter_uuid = self.module.params.get('vcenter_uuid')
+        parameters = self.module.params
+        target = self.get_target(parameters['targets'])
+        cluster_name = target['cluster']
+
+        if target['cluster']:
+            cluster_group_id, payload, new_host_id = self.process_cluster_target(target)
+        else:
+            cluster_name, cluster_group_id, payload, new_host_id = self.process_non_cluster_target(parameters)
+
+        if not isinstance(new_host_id, list):
+            new_host_id = [new_host_id]
+            host_service_tags = [host_service_tags]
+
+        if not self.is_update_job_allowed(vcenter_uuid, cluster_group_id, cluster_name):
+            return
+
+        if self.is_job_name_existing(vcenter_uuid, self.module.params.get('job_name')):
+            return
+
+        firmware_update_needed, before_no_change_dict, after_no_change_dict, before_dict, after_dict = self.is_firmware_update_needed(
+            vcenter_uuid, cluster_group_id, new_host_id, parameters['targets'], host_service_tags)
+
+        if self.module.check_mode or self.module.params.get('_ansible_check_mode'):
+            self.handle_check_mode(firmware_update_needed, before_no_change_dict,
+                                   after_no_change_dict, before_dict, after_dict)
+
+        if firmware_update_needed:
+            self.handle_firmware_update(vcenter_uuid, cluster_group_id, payload, parameters,
+                                        before_dict, after_dict)
+
+        else:
+            self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG,
+                                  diff={"before": before_no_change_dict,
+                                        "after": after_no_change_dict}, changed=False)
+
+    def process_cluster_target(self, target):
+        """
+        Process the cluster target.
+
+        Args:
+            target (dict): The target cluster.
+
+        Returns:
+            tuple: A tuple containing the cluster group ID, payload, and new host ID.
+        """
+        vcenter_uuid = self.module.params.get('vcenter_uuid')
+        host_ids, host_service_tags = self.get_host_id(vcenter_uuid, target)
+        if host_ids is None or not host_ids:
+            self.module.exit_json(msg=CLUSTER_HOST_NOT_FOUND_MSG, skipped=True)
+        cluster_name = target['cluster']
+        cluster_group_id = self.omevv_info_obj.get_group_id_of_cluster(vcenter_uuid,
+                                                                       cluster_name)
+        payload = self.get_payload_details(host_id=host_ids)
+        new_host_id = host_ids
```

Comment:
new_host_id is being only used in return, remove this line

Reply:
Incorporated the comment